### PR TITLE
Don't crash when RenderStyle is NULL for elements like optgroup when

### PR DESCRIPTION
**Pull request by chirags27**
#### 324460324818d8c11892d690a58d5b7b37b2209d
<pre>
Don&apos;t crash when RenderStyle is NULL for elements like optgroup when
rendering
<a href="https://bugs.webkit.org/show_bug.cgi?id=248575">https://bugs.webkit.org/show_bug.cgi?id=248575</a>

Reviewed by Simon Fraser.

* LayoutTests/fast/rendering/render-style-null-optgroup-crash-expected.txt: Added.
* LayoutTests/fast/rendering/render-style-null-optgroup-crash.html: Added.
* Source/WebCore/rendering/RenderListBox.cpp:
(WebCore::RenderListBox::paintItemForeground):
(WebCore::RenderListBox::paintItemBackground):

Canonical link: <a href="https://commits.webkit.org/257295@main">https://commits.webkit.org/257295@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/75254e48c615bdbb4ba21393158dc0c3a8e32fb4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/98324 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/7531 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/31461 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/107772 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/168061 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/102265 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8036 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/36290 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/90898 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/104403 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/103977 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/6096 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/84914 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33122 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/87926 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/89660 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/76068 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/1487 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/21036 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/1430 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/22562 "Found 1 new test failure: streams/readable-stream-default-controller-error.html (failure)") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5029 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6326 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/45050 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/2782 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41975 "Passed tests") | |

### CODE REVIEW

File: Source/WebCore/rendering/RenderListBox.cpp [hunk 1]
```diff
@@ -498,7 +501,10 @@ void RenderListBox::paintItemBackground(PaintInfo& paintInfo, const LayoutPoint&
 {
     const auto& listItems = selectElement().listItems();
     RefPtr listItemElement = listItems[listIndex].get();
-    auto& itemStyle = *listItemElement->computedStyle();
+    auto computedRenderStyle = listItemElement->computedStyle();
```
Comment (smfr) [inline Source/WebCore/rendering/RenderListBox.cpp]: Just keep the `itemStyle` name and use `itemStyle->` for the two usages later in the function.

File: LayoutTests/fast/rendering/render-style-null-optgroup-crash.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<style>
```
Comment (smfr) [inline LayoutTests/fast/rendering/render-style-null-optgroup-crash.html:1]: Can we simplify this testcase at all?

**Review by smfr — APPROVED** [review]

File: LayoutTests/fast/rendering/render-style-null-optgroup-crash.html (hunk 2 above)
Comment (chirags27) [reply LayoutTests/fast/rendering/render-style-null-optgroup-crash.html:1]: This is already the reduced version - Removing stuff any further from this doesn't reproduce the issue.

**Comment by chirags27** [comment]
The test failures don't seem related to the change in question, and are caused by an unrelated crash

**Pull request merged**